src/Interfaces/HandlerInterface.php

```diff
@@ -3,52 +3,9 @@
 
 namespace LoyaltyCorp\Search\Interfaces;
 
-interface HandlerInterface
+/**
+ * @deprecated Migrate to use EntitySearchHandlerInterface
+ */
+interface HandlerInterface extends EntitySearchHandlerInterface
```

Comment:
When we do BC interfaces, the BC interface needs to be the parent, because when we have someone depending on HandlerInterface in their code, it means our new handlers will not support that typehint.

Typically, you'd create EntitySearchHandlerInterface which extends HandlerInterface - and leave the methods on HandlerInterface until you're ready to remove it.

This way, when something typehints HandlerInterface, the current Handlers (that extend EntitySearchHandlerInterface) will be compatible.

Reply:
I think in this case though we can just say "sorry, no BC" and let pagey know he needs to update stuff if theres been any implementation done.

Reply:
I have removed the interface, but if there were existing search handlers implementing `HandlerInterface` wouldn't they be compatible with what was done? 

For example when populating an index the indexer checks if the type is of `EntitySearchHandlerInterface` and does the population, but if there are existing search handlers which implement `HandlerInterface` and our BC says

`EntitySearchHandlerInterface extends HandlerInterface`

won't the above check fail because HandlerInterface is not instance of EntitySearchHandlerInterface? but if HandlerInterface extends EntitySearchHandlerInterface this would all stay happy.

Reply:
What would happen if i had

```
public function doThingsWithHandler(HandlerInterface $handler) {}
```

In my code, and all of a sudden all the handlers I was depending on no longer implemented HandlerInterface but implemented EntitySearchHandlerInterface? They're no longer compatible

Reply:
ah in that case yes, so the BC would have only worked if I had not changed every other place in search package to not check for `EntitySearchHandlerInterface` but rather stay at `HandlerInterface`